read.go

```diff
@@ -124,3 +125,18 @@ func Parse(data []byte) (*Document, error) {
 	}
 	return &doc, nil
 }
+
+// normalizeEndings looks for instances of "\r\n" and flattens them to "\n".
+// If it finds no instances of "\r\n", the original byte slice is returned unchanged.
+//
+// This function does not bring joy; however,
+// see https://github.com/warpfork/go-testmark/pull/4#issuecomment-922760414
+// and see https://github.com/warpfork/go-testmark/pull/4#issuecomment-922782549
+// for discussion.  Performing this kind of normalization to data hunk boundaries
+// seems to be a "least bad" behavior in a practical sense.
+func normalizeEndings(in []byte) []byte {
+	if bytes.Count(in, sigilCrLf) == 0 {
+		return in
+	}
```

Comment:
I imagine `bytes.Replace` is already smart enough to return the input unchanged if there are no matches

Reply:
Oddly enough, it does specifically the opposite: it _always_ [returns a copy](https://cs.opensource.google/go/go/+/master:src/bytes/bytes.go;l=1015;drc=5baf60d47245c792c50a349cd6b8586d23204895).

(I initially had the same expectation you did!)

Reply:
Ah, it's documented to "return a copy", so that makes sense. Weird docs though.

Reply:
Agreed.